src/main/java/io/kestra/plugin/hightouch/AbstractHightouchConnection.java

```diff
@@ -0,0 +1,77 @@
+package io.kestra.plugin.hightouch;
+
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.Task;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.net.http.HttpRequest;
+import java.net.MalformedURLException;
+import java.net.ConnectException;
+import java.nio.charset.StandardCharsets;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kestra.core.serializers.JacksonMapper;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotNull;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class AbstractHightouchConnection extends Task {
```

Comment:
```suggestion
abstract class AbstractHightouchConnection extends Task {
```

You can use package private visibility as this class is only used by the Sync class